src/Tizen.NUI.Components/Controls/Adapters/LottieButtonAdapter.cs

```diff
@@ -0,0 +1,139 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+using System.ComponentModel;
+using Tizen.NUI.BaseComponents;
+
+namespace Tizen.NUI.Components
+{
+    /// <summary>
+    /// LottieButtonAdapter is a ButtonAdapter class that uses Lottie image for a Button icon.
+    /// </summary>
+    /// <remarks>
+    /// You can make a custom LottieButton by overriding methods.
+    /// </remarks>
+    /// <example>
+    /// <code>
+    /// public class CustomLottieButton : LottieButtonAdapter
+    /// {
+    ///     public CustomLottieButton(string url) : base(url)
+    ///     {
+    ///     }
+    /// 
+    ///     public override LottieFrameInfo? GetFrameInfoOnClick(bool isSelected)
+    ///     {
+    ///         return LottieFrameInfo.CreateAnimationRange(0, 30);
+    ///     }
+    /// }
+    /// </code>
+    /// </example>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public abstract class LottieButtonAdapter : ButtonAdapter, ILottieButtonAdapter
+    {
+        /// <summary>
+        /// The Lottie view that will be used as an icon part in a Button.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected LottieAnimationView LottieView { get; set; }
+
+        /// <summary>
+        /// A constructor that creates LottieButtonAdapter with a specified Lottie resource URL
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public LottieButtonAdapter(string resourceURL) : base()
+        {
+            LottieView = new LottieAnimationView();
+            LottieView.URL = resourceURL;
+            LottieView.StopBehavior = LottieAnimationView.StopBehaviorType.MaximumFrame;
+        }
+
+        /// <inheritdoc/>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override void OnCreateIcon(Button button, ref ImageView icon, ImageViewStyle style)
+        {
+            // Replace icon to the Lottie view
+            icon = LottieView;
+            style.WidthResizePolicy = ResizePolicyType.FillToParent;
+            style.HeightResizePolicy = ResizePolicyType.FillToParent;
+            GetFrameInfoOnCreate(button.IsSelected)?.Show(LottieView);
+        }
+
+        /// <inheritdoc/>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override void OnControlStateChanged(Button button, ControlStates previousState, ControlStates currentState, bool byUI, Touch touchInfo)
+        {
+            HandleStateChange(currentState, button.IsSelected, byUI, this, LottieView);
+        }
+
+        /// <inheritdoc/>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public virtual LottieFrameInfo? GetFrameInfoOnCreate(bool isSelected)
```

Comment:
as I know, class is nullable. so if LottieFrameInfor is class, then ? is not needed, I think...